src/main.rs

```diff
@@ -9,69 +9,166 @@ mod utils;
-  
     /// Disable a postprocessor
-    /// Available postprocessors: shfmt, bshchk
-    /// To select multiple, pass this argument multiple times with different values.
-    /// This argument also supports a wilcard match, like "*" or "s*mt"
+    /// Available postprocessors: 'shfmt', 'bshchk'
+    /// To select multiple, pass multiple times with different values
+    /// Argument also supports a wilcard match, like "*" or "s*mt"
     #[arg(long, verbatim_doc_comment)]
     no_proc: Vec<String>,
 
-    /// Minify the resulting code
+    /// Minify the output file
     #[arg(long)]
     minify: bool,
 }
 
+#[derive(Args, Clone, Debug)]
+struct DocCommand {
+    /// Input filename ('-' to read from stdin)
+    input: PathBuf,
+
+    /// Output directory (relative to input file, default 'docs')
+    output: Option<PathBuf>,
+
+    /// Show standard library usage in documentation
+    #[arg(long)]
+    usage: bool,
+}
+
 fn main() -> Result<(), Box<dyn Error>> {
     let cli = Cli::parse();
-    if cli.docs {
-        handle_docs(cli)?;
-    } else if let Some(ref code) = cli.eval {
-        handle_eval(code.to_string(), cli)?;
-    } else {
-        handle_compile(cli)?;
+    if let Some(command) = cli.command {
+        match command {
+            CommandKind::Eval(command) => {
+                handle_eval(command)?;
+            }
+            CommandKind::Run(command) => {
+                let options = CompilerOptions::default();
+                let (code, messages) = compile_input(command.input, options);
+                execute_output(code, command.args, messages)?;
+            }
+            CommandKind::Check(command) => {
+                let options = CompilerOptions {
+                    no_proc: command.no_proc.clone(),
+                    minify: false,
+                };
+                compile_input(command.input, options);
+            }
+            CommandKind::Build(command) => {
+                let output = create_output(&command);
+                let options = CompilerOptions {
+                    no_proc: command.no_proc.clone(),
+                    minify: command.minify,
+                };
+                let (code, _) = compile_input(command.input, options);
+                write_output(output, code);
+            }
+            CommandKind::Doc(command) => {
+                handle_doc(command)?;
+            }
+            CommandKind::Comp => {
+                handle_complete();
+            }
+        }
+    } else if let Some(input) = cli.input {
+        let options = CompilerOptions::default();
+        let (code, messages) = compile_input(input, options);
+        execute_output(code, cli.args, messages)?;
     }
     Ok(())
 }
 
-fn handle_compile(cli: Cli) -> Result<(), Box<dyn Error>> {
-    let input = if let Some(input) = cli.input.clone() {
-        String::from(input.to_string_lossy().trim())
+fn create_output(command: &BuildCommand) -> PathBuf {
+    if let Some(output) = &command.output {
```

Comment:
Note to self: if the user enters `cat args.ab | amber build -`, we need to return `-` instead of `-.sh` here, so we write to standard output.